From dbaryshkov@gmail.com  Fri Dec 21 10:35:55 2007
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Date: Fri, 21 Dec 2007 12:18:01 +0300
Subject: USB: tosa_udc_use_gpio_vbus.patch
To: Greg KH <greg@kroah.com>
Message-ID: <20071221091801.GA6789@doriath.ww600.siemens.net>
Content-Disposition: inline

From: Dmitry Baryshkov <dbaryshkov@gmail.com>

Use gpio_vbus instead of udc_is_connected for udc on tosa.

Signed-off-by: Dmitry Baryshkov <dbaryshkov@gmail.com>
Acked-by: Russell King <rmk+kernel@arm.linux.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/arm/mach-pxa/tosa.c |    9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

--- a/arch/arm/mach-pxa/tosa.c
+++ b/arch/arm/mach-pxa/tosa.c
@@ -157,15 +157,10 @@ static void tosa_udc_command(int cmd)
 	}
 }
 
-static int tosa_udc_is_connected(void)
-{
-	return ((GPLR(TOSA_GPIO_USB_IN) & GPIO_bit(TOSA_GPIO_USB_IN)) == 0);
-}
-
-
 static struct pxa2xx_udc_mach_info udc_info __initdata = {
 	.udc_command		= tosa_udc_command,
-	.udc_is_connected	= tosa_udc_is_connected,
+	.gpio_vbus		= TOSA_GPIO_USB_IN,
+	.gpio_vbus_inverted	= 1,
 };
 
 /*
